euphonic/design/05_powder_averaging_discussion.md

```diff
@@ -0,0 +1,318 @@
+  the project we should get a baseline method in place and then
+  experiment from there, rather than designing everything in theory
+  first.
+- Fundamentally the goal of the sampling exercise is to perform a
+  numerical integration. In the numerical methods field there has been
+  a lot of work on integration problems, including methods which use
+  adaptive step sizes to obtain a consistent uncertainty when
+  integrating arbitrarily complex functions.
+- This function is not arbitrary and there should be some useful
+  information lurking in the force constants data.
+
+
+### Is the same sampling mesh suitable for the **A** matrix used in Debye-Waller calculation, or should this be a separate q-mesh?
+
+- This point was not discussed in the meeting. It's worth observing
+  that DW should only need sampling within a single Brillouin zone,
+  while S reaches out further in reciprocal space. We'll probably stick with
+  Monkhorst-Pack grids initially.
+
+
+### Can we use statistical methods to quantify uncertainty in the convergence?
+
+- Statistical methods, propogation of error approaches could provide
+  some useful tools
+- Experimentally we can examine uncertainty by comparing different
+  detector contributions to the same bin
+- We should be able to somehow examine the changing variance in S as more
+  **q**-points are sampled.
+
+<a id="org72cee07"></a>
+
+## Existing approaches
+
+
+<a id="orgce0cfe3"></a>
+
+### Random sampling
+
+-   This has been mentioned in PACE meetings. How much has it been
+    used?
+    - Random sampling on spherical shells seems more common that fully
+      random sampling in **q**-space.
+-   Has natural advantage of being uniform in space... eventually.
+-   Convergence can be examined systematically, adding new points
+    without removing old ones. A smoother process than replacing
+    entire sampling meshes with different criteria.
+-   Strictly we can sample (**q**,ω) randomly, *or* sample **q** randomly and search ω systematically at
+    each **q**.
+    - The former approach is applicable to codes like MDANSE that
+      sample S(**q**,ω) in a response formalism from dynamic data.
+    - The latter approach maps better to PACE; at each **q** we go to
+      the force constants and get all the active ω. This has the
+      advantage that every sample should find some signal, without the
+      risk of hitting "empty space".
+
+
+<a id="orgd1b667c"></a>
+
+### Spherical sampling
+
+-   SpinW seems to use a Fibonacci-related method for uniform
+    sampling at spherical surfaces
+    -   points per shell has to be a Fibonacci number
+        - That's not a deal-breaker but could be annoying when exploring convergence
+    -   or can sample *randomly* at *each* spherical shell surface
+-   MDANSE requires the user to set up a q lattice for sampling,
+    including a "spherical grid" option with a set range of "shells"
+    and "vectors".
```

Comment:
Not sure if this is similar to MDANSE, but `mcphase` does a very primitive *regular* spherical grid (specified by number of points along theta and phi directions and the by number of abs(q) bins): https://github.com/mducle/mcphase/blob/master/bin/powdermagnon.pl#L225 

This is *not* the best way!